Feature/notifications

lsh9511 opened the pull request:
## ✅ PR 요약
- 작업 요약: Notifications 앱 생성, 모델 정의 및 세팅

## 📄 상세 내용
- apps의 app_notifications 폴더 프로젝트 구조에 맞게 생성
- notifications 모델 ERD기반 정의
- 폴더 구조 정리 및 마이그레이션 생성 -> 마이그레이트

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

lsh9511 committed:
✨ feat : app_notifications 생성 및 폴더링

lsh9511 committed:
✨ feat : notifications 모델 정의 및 settings.py notification 앱 등록

# Conflicts:
#	config/settings/base.py

lsh9511 committed:
✨ feat : notifications 모델 정의 및 settings.py notification 앱 등록

# Conflicts:
#	config/settings/base.py

lsh9511 committed:
🎨 style    : 코드 포매팅 수정

lsh9511 committed:
🎨 style    : 코드 포매팅 수정

lsh9511 committed:
🚚 build    : views.py 삭제,models.py models디렉토리 이동 및 설정코드 추가 , 마이그레이션

lsh9511 committed:
🎨 style    : 코드 포매팅 수정 및 타입체킹

2joonkim commented on apps/notifications/models/notifications_models.py:
```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth import get_user_model
```
테이븖 명 컨벤션 참고하시고 영어로 통일 부탁드리겠습니다!

lsh9511 replied:
테이블명 컨벤션이 장고 관리자페이지상 나오는 테이블명 수정 말씀하시는건가요?!
    class Meta:
        verbose_name = "알림"
        verbose_name_plural = "알림 목록"
        이거 영어로 수정해두면 될까요?

choismjames23 commented on apps/app_notifications/models/notifications_models.py:
```diff
@@ -0,0 +1,50 @@
+
+    class NotificationType(models.TextChoices):
+        APPLICATION_CREATED = "APPLICATIONS_CREATED", "공고 지원 알림"
+        APPLICATION_STATUS_CHANGED = "APPLICATION_STATUS_CHANGED", "지원 승인/거절 알림"
```
API 명세서 보니까 승인 / 거절 알림 부분을 분리해서 작성하셨던데, 알림 타입에도 구분해서 적어두는 것이 좋아보입니다!

lsh9511 replied:
이해했습니다 감사합니다

lsh9511 committed:
🚚 build    : NotificationType 승인/거절 알림 타입 나눔

lsh9511 committed:
🚚 build    : 관리자페이지 테이블명 영어로 수정

Meoyoug reviewed (changes requested):
충돌 해결 해주시고, admin.py  파일은 삭제후에 재 푸시 부탁드려요~!

lsh9511 committed:
📝 docs     : admin.py 삭제

lsh9511 commented:
테스트 끝나면 충돌 해결후 머지하면 될까용?

Meoyoug commented:
> 테스트 끝나면 충돌 해결후 머지하면 될까용?

충돌해결은 로컬에서 최신화하시면서 해주시고 푸시해주세요~

lsh9511 committed:
🚚 build    : rebase후 app_notifications apps 추가

# Conflicts:
#	config/settings/base.py

# Conflicts:
#	config/settings/base.py

# Conflicts:
#	config/settings/base.py

# Conflicts:
#	config/settings/base.py

lsh9511 commented:
로컬 수정해서 push 했는데 conflict가 납니당 혹시 확인좀 부탁드려도될까요

Meoyoug commented:
> 로컬 수정해서 push 했는데 conflict가 납니당 혹시 확인좀 부탁드려도될까요

base.py만 수정한다고 충돌이 안나는게 아니라 현재 이 feature 브랜치에서 git rebase develop 을 하면 분명히 충돌이 날텐데 거기서 충돌해결을 해야죠

lsh9511 committed:
🚚 build    : models.py verbose_name 수정

lsh9511 committed:
💡 chore    : notifications 앱 이름 변경

lsh9511 committed:
🎨 style : 코드 포매팅

lsh9511 committed:
🚚 build    : notifications 마이그레이션

lsh9511 committed:
🐛 fix: base.py 코드 포매팅, models.py 삭제
